wgsl/index.bs

```diff
@@ -8605,11 +8624,192 @@ The following algorithm describes how to compute these tags for a given function
+          where *p* is a [=let-declaration=] initialized to &*x*
+  <tr><td class='nowrap'>
+          *f*(*x*) = *e*;<br>
+      <td rowspan=3>*Vout*(*s*)
+      <td rowspan=3>*V*(*e*), *V*(*prev*)
+
+          Note: This is a [=partial assignment=] to *x*.
+
+          Note: Partial assignments include the previous value since only a subset of components are updated.
+  <tr><td class='nowrap'>
+          **p* = *e*;<br>
+          where *p* is a [=let-declaration=] initialized to &*f*(*x*)<br>
+  <tr><td class='nowrap'>
+          **f*(*p*) = *e*;<br>
+          where *p* is a [=let-declaration=] initialized to &*f*(*x*) or &*x*
+  <tr><td>*s1* *s2*<br>
+          where *Next* is in behavior of *s1*.
+
+          Note: *s1* often ends in a semicolon.
+      <td>*Vin*(*s2*)
+      <td>*Vout*(*s1*)
+  <tr><td class='nowrap'>
+          if *e* *s1* else *s2*<br>
+          where *Next* is in the behaviors of both *s1* and *s2*
+      <td>*Vin*(*next*)
+      <td>*Vout*(*s1*), *Vout*(*s2*)
+  <tr><td class='nowrap'>
+          if *e* *s1* else *s2*<br>
+          where *Next* is in the behavior of *s1*, but not *s2*
+      <td>*Vin*(*next*)
+      <td>*Vout*(*s1*)
+  <tr><td class='nowrap'>if *e* *s1* else *s2*<br>
+          where *Next* is in the behavior of *s2*, but not *s1*
+      <td>*Vin*(*next*)
+      <td>*Vout*(*s2*)
+  <tr><td>loop { *s1* continuing { *s2* } }
+      <td>*Vin*(*s1*)
+      <td>*Vout*(*prev*), *Vout*(*s2*)
+  <tr><td>loop { *s1* continuing { *s2* } }
+      <td>*Vin*(*s2*)
+      <td>*Vout*(*s1*),<br>
+          *Vout*(*s*<sub>i</sub>)<br>
+          for all *s*<sub>i</sub> in *s1* whose behavior is {*Continue*} and transfer control to *s2*
+  <tr><td>loop { *s1* continuing { *s2* } }
+      <td>*Vin*(*next*)
+      <td>*Vout*(*s2*),<br>
+          *Vout*(*s*<sub>i</sub>)<br>
+          for all *s*<sub>i</sub> in *s1* whose behavior is {*Break*} and transfer control to *next*
+  <tr><td>switch *e* {<br>
+            case _: *s1*<br>
+            case _: *s2*<br>
+            ...<br>
+            case _: *s3*<br>
+          }<br>
+          where *Fallthrough* is not in the behavior of *s*<sub>i-1</sub>
+      <td>*Vin*(*s*<sub>i</sub>)
+      <td>*Vout*(*prev*)
+  <tr><td>switch *e* {<br>
+            case _: *s1*<br>
+            case _: *s2*<br>
+            ...<br>
+            case _: *s3*<br>
+          }<br>
+          where *Fallthrough* is in the behavior of *s*<sub>i-1</sub>
+      <td>*Vin*(*s*<sub>i</sub>)
+      <td>*Vout*(*prev*), *Vout*(*s*<sub>i-1</sub>)
+  <tr><td>switch *e* {<br>
+            case _: *s1*<br>
+            case _: *s2*<br>
+            ...<br>
+            case _: *s3*<br>
+          }<br>
+      <td>*Vin*(*next*)
+      <td>*Vout*(*s*<sub>i</sub>),<br>
+          for all *s*<sub>i</sub> whose behaviour is {*Break*}, and<br>
+          *Vout*(*s*<sub>j</sub>)<br>
+          for all statements in *s*<sub>j</sub> whose behavior is {*Break*} and trasfer control to *next*
```

Comment:
?  This seems duplicated?

Isn't it      Vout(si) for all whos beahviour includes either Break or Next.     

e.g. 

   switch(c){ 
      case 1:  {  x= 1; break; }
      default:  {  x = 2; } // behaivour Next, should still be included.
  }

Reply:
So the first set (dealing with *s<sub>i</sub>*) are values for the overall compound statements for each case clause. The second set (dealing with *s<sub>j</sub>*) are for values that are determined due to statements within each compound statement that break.

For example:
```
switch (c2) {
  case 1: { x = 1; }
  default: { x = 2; if (c2) { x = 3; break; } }
}
```

You're right that the first set should include *Next* as a condition. The second set covers adding the edge that covers the assignment of 3 to x.

Reply:
Ah, for the second part it's statements *inside* s_j that break.

Editorial:  Suggest replacing "in" with "inside".  Saying "in" seems like your indexing the Sj's themselves.  Or at least that's how I read it.